src/apps/lwaftr/ndp.lua

```diff
@@ -151,6 +151,12 @@ function is_neighbor_solicitation_for_ips(pkt, local_ips)
    return false
 end
 
+local function to_ether_addr(pkt, offset)
+   local ether_src = ffi.new("uint8_t[?]", 6)
+   ffi.copy(ether_src, pkt.data + offset, 6)
```

Comment:
OK, I can return a pointer, but I think at a later point it's necessary to make a copy into `o.dst_eth`, since the packet is immediately freed once `o.dst_eth` is assigned.

https://github.com/Igalia/snabbswitch/blob/lwaftr_starfruit/src/apps/lwaftr/ipv6_apps.lua#L201

If it stores a pointer, and the packet is freed, on a later use will point to garbage. WDYT?